src/v2/Apps/Fair/Components/FairTiming.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react"
+import { Sans, Text } from "@artsy/palette"
+import { createFragmentContainer, graphql } from "react-relay"
+import { FairTiming_fair } from "v2/__generated__/FairTiming_fair.graphql"
+import { WithCurrentTime } from "v2/Components/WithCurrentTime"
+import { DateTime, Duration } from "luxon"
+
+interface FairMetaProps {
+  fair: FairTiming_fair
+}
+
+const SEPARATOR = <>&nbsp;:&nbsp;</>
+
+const FairTiming: React.FC<FairMetaProps> = ({
+  fair: { exhibitionPeriod, startAt, endAt },
+}) => {
+  return (
+    <>
+      <Text variant="text" color="black80">
+        {exhibitionPeriod}
+      </Text>
+      <Text variant="text" color="black60">
+        <WithCurrentTime syncWithServer>
+          {currentTime => {
+            const durationTilEnd = Duration.fromISO(
+              DateTime.fromISO(endAt)
+                .diff(DateTime.fromISO(currentTime))
+                .toString()
+            )
+            const daysTilEnd = durationTilEnd.as("days")
+            const secondsTilEnd = durationTilEnd.as("seconds")
+
+            const hasStarted =
+              Duration.fromISO(
+                DateTime.fromISO(startAt)
+                  .diff(DateTime.fromISO(currentTime))
+                  .toString()
+              ).seconds < 0
+            const closesSoon = daysTilEnd <= 3 && daysTilEnd > 1
+            const hasEnded = Math.floor(secondsTilEnd) <= 0
+            const closesToday = daysTilEnd < 1 && !hasEnded
+
+            return (
+              <Sans size="3" weight="medium">
+                {hasEnded && "Closed"}
+                {!hasStarted && "Opening Soon"}
+                {closesSoon && `Closes in ${daysTilEnd} days`}
+                {closesToday && (
+                  <>
+                    Closes in{" "}
+                    {Math.max(0, Math.floor(durationTilEnd.as("hours") % 24))}
+                    {SEPARATOR}
+                    {Math.max(0, Math.floor(durationTilEnd.as("minutes") % 60))}
+                    {SEPARATOR}
+                    {Math.max(0, Math.floor(secondsTilEnd % 60))}
+                  </>
+                )}
```

Comment:
It does seem like the body of this can just be abstracted into a hook which would make it more resusable and simpler to test.

Reply:
Yea, for ease of testing (plus the reuse for show), it might make sense to extract it out now.